File: SDKs/github.com/ghes-3.0/1.1.4/go/pkg/models/operations/actionslistorgsecrets.go
package operations

import (
"openapi/pkg/models/shared")

type ActionsListOrgSecretsPathParams struct {
    Org string `pathParam:"style=simple,explode=false,name=org"`
    
}

type ActionsListOrgSecretsQueryParams struct {
    Page *int64 `queryParam:"style=form,explode=true,name=page"`
    PerPage *int64 `queryParam:"style=form,explode=true,name=per_page"`
    
}

type ActionsListOrgSecretsRequest struct {
    PathParams ActionsListOrgSecretsPathParams 
    QueryParams ActionsListOrgSecretsQueryParams 
    
}

type ActionsListOrgSecrets200ApplicationJSON struct {
    Secrets []shared.OrganizationActionsSecret `json:"secrets"`
    TotalCount int64 `json:"total_count"`
    
}

type ActionsListOrgSecretsResponse struct {
    ContentType string 
    Headers map[string][]string 
    StatusCode int64 
    ActionsListOrgSecrets200ApplicationJSONObject *ActionsListOrgSecrets200ApplicationJSON 
    
}

File: SDKs/github.com/ghes-2.22/1.1.4/typescript/src/sdk/models/operations/actionslistorgsecrets.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ActionsListOrgSecretsPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=org" })
  org: string;
}


export class ActionsListOrgSecretsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=per_page" })
  perPage?: number;
}


export class ActionsListOrgSecrets200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=secrets", elemType: shared.OrganizationActionsSecret })
  secrets: shared.OrganizationActionsSecret[];

  @SpeakeasyMetadata({ data: "json, name=total_count" })
  totalCount: number;
}


export class ActionsListOrgSecretsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ActionsListOrgSecretsPathParams;

  @SpeakeasyMetadata()
  queryParams: ActionsListOrgSecretsQueryParams;
}


export class ActionsListOrgSecretsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  headers: Map<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  actionsListOrgSecrets200ApplicationJsonObject?: ActionsListOrgSecrets200ApplicationJson;
}
